google-http-client-xml/src/test/java/com/google/api/client/xml/AtomTest.java

```diff
@@ -44,8 +75,205 @@ public void subtestSetSlugHeader(HttpHeaders headers, String expectedValue, Stri
+   * Manuel tests of a constant string to see if the data structure can be parsed in the regular way
+   * and get the same result.
+   *
+   * The purpose of this test is to evaluate, if the parsed elements are the same with the {@link
+   * AtomFeedParser}.
+   */
+  @SuppressWarnings("unchecked")
+  @Test
+  public void testAtomFeedParserRegualar() throws Exception {
+    Feed feed = new Feed();
+    XmlPullParser parser = Xml.createParser();
+    parser.setInput(new StringReader(SAMPLE_FEED));
+    XmlNamespaceDictionary namespaceDictionary = new XmlNamespaceDictionary();
+    Xml.parseElement(parser, feed, namespaceDictionary, null);
+    assertNotNull(feed);
+    assertEquals(2, feed.entry.length);
+
+    assertEquals("John Doe", feed.author.name);
+    assertEquals("urn:uuid:60a76c80-d399-11d9-b93C-0003939e0af6", feed.id);
+    assertEquals("2003-12-13T18:31:02Z", feed.updated);
+    assertEquals("Example Feed", feed.title);
+    assertEquals("http://example.org/", feed.link.href);
+
+    FeedEntry entry1 = feed.entry[0];
+    //assertNotNull(feed.entry);
+    assertEquals("urn:uuid:1225c695-cfb8-4ebb-aaaa-80da344efa6a", entry1.id);
+    assertEquals("2003-12-13T18:30:02Z", entry1.updated);
+    assertEquals("Some text.", entry1.summary);
+    assertEquals("Atom-Powered Robots Run Amok", entry1.title);
+    assertEquals("http://example.org/2003/12/13/atom03", entry1.link.href);
+
+    FeedEntry entry2 = feed.entry[1];
+    assertEquals("urn:uuid:1225c695-cfb8-4ebb-aaaa-80da344efa62", entry2.id);
+    assertEquals("2003-12-13T18:32:02Z", entry2.updated);
+    assertEquals("Some other text.", entry2.summary);
+    assertEquals("Atom-Powered Robots Run Amok!", entry2.title);
+    assertEquals("http://example.org/2003/12/13/atom02", entry2.link.href);
+  }
+
+  /**
+   * Reading an XML ATOM Feed from a file and valid if all the {@link FeedEntry} are present. No
+   * detailed evaluation of each element
+   *
+   * The purpose of this test is to read a bunch of elements which contain additional elements
+   * (HTML in this case), that are not part of the {@link FeedEntry} and to see if there is an issue
+   * if we parse some more entries.
+   */
+  @SuppressWarnings("unchecked")
+  @Test
+  public void testHeiseFeedParser() throws Exception {
+    XmlPullParser parser = Xml.createParser();
+    String read = readFile("heise-atom.xml");
+    parser.setInput(new StringReader(read));
+    XmlNamespaceDictionary namespaceDictionary = new XmlNamespaceDictionary();
+    AbstractAtomFeedParser atomParser = new AtomFeedParser<Feed, FeedEntry>(namespaceDictionary,
+        parser, new ByteArrayInputStream(read.getBytes()), Feed.class, FeedEntry.class);
+    Feed feed = (Feed) atomParser.parseFeed();
+    assertNotNull(feed);
+
+    int counter = 0;
+    while (atomParser.parseNextEntry() != null) {
+      counter++;
+    }
+
+    atomParser.close();
+    assertEquals(62, counter);
+  }
+
+  /**
+   * We need to method to read the file into a string
```

Comment:
Don't use first person in javadoc. Consider, "read the file into a string". But you should probably just use Guava and

URL url = Resources.getResource(file);
String text = Resources.toString(url, Charsets.UTF_8);

Reply:
Definitely more elegant than the current implementation, will use guava.